tests/core/chain-object/test_chain_reorganization.py

```diff
@@ -127,28 +79,42 @@ def test_import_block_with_reorg(chain, fork_chain, funded_address_private_key):
 
 def test_import_block_with_reorg_with_current_head_as_uncle(
         chain,
-        fork_chain,
         funded_address_private_key):
     """
     https://github.com/ethereum/py-evm/issues/1185
     """
-    # mine a block on the main chain which will eventually become an uncle on
-    # the main chain after a reorg.
-    block = chain.mine_block()
-
-    # Force the fork_chain to diverge from the main chain
-    fork_chain.header = fork_chain.header.copy(extra_data=b'fork-it!')
-    f_block_a = fork_chain.mine_block()
-
-    # now mine a block which has the current chain head as an uncle.
-    assert f_block_a != block
-    f_block_b = fork_chain.mine_block(uncles=(block.header,))
-
-    # ensure that we don't cause a re-org with our first import.
-    _, new_chain, _ = chain.import_block(f_block_a)
-    assert new_chain == tuple()
+    def converge_split_fn(results):
+        chain_a = results[0][-1]
+        chain_b = results[1][-1]
+
+        # the chain head from the main chain will become an uncle once we
+        # import the fork chain blocks.
+        uncle = chain_a.get_canonical_head()
+
+        pipe(
+            chain_b,
+            api.mine_block(uncles=(uncle,)),
+        )
+        head = chain_b.get_canonical_head()
+        block_a = chain_b.get_canonical_block_by_number(head.block_number - 1)
+        block_b = chain_b.get_canonical_block_by_number(head.block_number)
+
+        return pipe(
+            chain_a,
+            api.import_block(block_a),
+            api.import_block(block_b),  # the block with the uncle.
+        )
+
+    chain = pipe(
+        chain,
+        api.chain_split(
+            (api.mine_block(),),  # this will be the 'uncle'
+            (api.mine_block(extra_data=b'fork-it'),),
+            exit_fn=converge_split_fn,
```

Comment:
I'm not feeling this `exit_fn` API still. If you aren't a fan of this: https://github.com/ethereum/py-evm/pull/1209#discussion_r212447203

Another option is to not use the pipe at all when chain splitting, and have it be its own top-level method:

```py
chain_a, chain_b = api.chain_split(
        chain,
        (
            (api.mine_block(),),  # this will be the 'uncle'
            (api.mine_block(extra_data=b'fork-it'),),
         ),
)

## insert the body of convert_split_fn here
```

Reply:
I'm not in love with `exit_fn` either.  I'm going to go with your suggestion to just have `chain_split` return the resulting chain for all of the chains.

This however is an API I want to revisit at some point because I **really** want a way to concisely, explicitly cause a fork, and then adopt the fork blocks as uncles into the main fork and to have it still be easy to read and understand what is going on.  For now, this will do.